test/service/test_manifest_async.py

```diff
@@ -185,75 +187,102 @@ def test(self, mock_uuid, mock_helper):
-                                                                    is_last=False,
-                                                                    file_name=None,
-                                                                    config=None,
-                                                                    multipart_upload_id=None,
-                                                                    part_etags=None,
-                                                                    page_index=None,
-                                                                    is_last_page=None,
-                                                                    search_after=None))
-                            )
-                            mock_helper.describe_execution.assert_not_called()
-                            mock_helper.reset_mock()
-                            mock_helper.describe_execution.return_value = {'status': 'RUNNING'}
-                            params = None
-                        elif i == 1:
-                            mock_helper.start_execution.assert_not_called()
-                            mock_helper.describe_execution.assert_called_once()
-                            mock_helper.reset_mock()
-                            # simulate absence of output due eventual consistency
-                            mock_helper.describe_execution.return_value = {'status': 'SUCCEEDED'}
-                        elif i == 2:
-                            mock_helper.start_execution.assert_not_called()
-                            mock_helper.describe_execution.assert_called_once()
-                            mock_helper.reset_mock()
-                            mock_helper.describe_execution.return_value = {
-                                'status': 'SUCCEEDED',
-                                'output': json.dumps(
-                                    {
-                                        'manifest': {
-                                            'location': str(manifest_url),
-                                            'was_cached': False,
-                                            'format_': format_.value,
-                                            'catalog': self.catalog,
-                                            'filters': filters,
-                                            'object_key': object_key
-                                        }
-                                    }
+
+                        def mock_get_manifest():
+                            if i == 1:
+                                p = partition.with_upload('some_upload_id')
+                                return p.next('some_etag')
+                            elif i == 2:
+                                return manifest
+
+                        with mock.patch.object(ManifestService,
+                                               'get_manifest',
+                                               return_value=mock_get_manifest()):
```

Comment:
`mock_get_manifest` isn't a mock because it's called when the patch is set up, not when the patch is active. Because of that and because it is only called once, there is no point in having a nested function here. I [tried to point this out earlier](https://github.com/DataBiosphere/azul/pull/3112#discussion_r658361772), using admittedly terse language, but you :+1:-ed it. If my comments aren't clear, please bring that up in PL.